### DISCUSSION TITLE
Stuck at "Resetting game objects"

### DISCUSSION THREAD

**Discussion by afdallah** [Pryaxis/TShock#3035] [General]
Hello everyone, I've tried running TShock on Ubuntu 24 and also on Docker. Both instances always hang at "Resetting game objects". 

They are hosted on AWS free tier instances with the following specs: **1 vCPU, 1 GiB memory, and 20 GB storage.**
<img width="1353" alt="image" src="https://github.com/Pryaxis/TShock/assets/11598084/295ac809-6aa4-4b7a-bc3e-dbc80b8bd796">

Please suggest what I can do to fix this. thanks.

**Comment by sgkoishi** [comment]
Probably you need more memory. You may also try the CLI flag `-c` (for TShock, not docker).

**Reply by hakusaro** [reply]
I concur with this assessment, 1gb is pretty difficult.

**Reply by afdallah** [reply]
Yes, `-c` flag did the trick. I also try to bump up the memory before and it's working. So yes, I guess it's memory issue. 
@sgkoishi  @hakusaro what is the recommended minimum spec for this to work stable?

**Reply by hakusaro** [reply]
Generally Terraria need at least a gigabyte (dedicated) without using a different tile handler. I think we've usually pushed people to 2gb but with different tile data storage, people have definitely run in more constrained environments.

**Reply by afdallah** [reply]
@hakusaro : I see, is there any downside if we use TShock's tile provider instead of the stock from Terraria?

**Reply by hakusaro** [reply]
Not really!

**Reply by sgkoishi** [reply]
The `-c` flag uses [Constileation](https://ikebukuro.tshock.co/#/tile-providers?id=constileation). It _might_ be slower but not actually noticeable, and in some cases even faster.

**Reply by afdallah** [reply]
Okay, sounds good.
One more thing, I often see the npc / slime teleporting or flying at random times. Do you guys know what potentially cause this? is it unstable network or something else

**Reply by sgkoishi** [reply]
When they teleport forwards, it's your client getting CPU bottleneck. Backwards means your server's CPU or network is poor.
For networking 0.5-1Mbps per player is usually enough. For CPU you can try 2vCPUs, but Terraria is not optimized to use multiple CPU cores so it's probably useless to upgrade to more than 4.
If you already have lots of CPUs and network throughput but still don't work well, try other hosting platforms.
